src/nvim/normal.c

```diff
@@ -1492,6 +1548,7 @@ bool do_mouse(oparg_T *oap, int c, int dir, long count, bool fixindent)
   int jump_flags = 0;           // flags for jump_to_mouse()
   pos_T start_visual;
   bool moved;                   // Has cursor moved?
+  bool in_winbar;               // mouse in window bar
   bool in_status_line;          // mouse in status line
```

Comment:
mouse can never be in both, maybe a mask or an int is appropriate?

```
in_statusline = 1 // statusline
in_statusline = 2 // winbar
in_statusline = 3 // tabline

Reply:
Idk how I'd integrate tabline into it since `in_tab_line` is static whereas the other two are not. And I personally think they're better off as separate booleans tbh

Reply:
`in_winbar` is just this value `(jump_flags & MOUSE_WINBAR)` so I think making this change defeats the point of this variable and using the value `(jump_flags & MOUSE_WINBAR)` is better.

Reply:
> `in_winbar` is just this value `(jump_flags & MOUSE_WINBAR)` so I think making this change defeats the point of this variable and using the value `(jump_flags & MOUSE_WINBAR)` is better.

Same with `in_status_line`, which is jsut `(jump_flags & IN_STATUS_LINE)`

Reply:
oops, I thought this was part of a struct. I see now it is only local vars.